### PR TITLE
Ds 1284 phenodigm rewrite

### PR DESCRIPTION
**Pull request by DSuveges**
This is the expected output:
```json
{
  "targetFromSourceId": "ENSG00000115977",
  "diseaseFromSourceMappedId": "Orphanet_79235",
  "resourceScore": "0.900900000000000",
  "datasourceId": "phenodigm",
  "datatypeId": "animal_model",
  "diseaseFromSource": "Crigler-Najjar Syndrome, Type Ii",
  "targetInModel": "ENSMUSG00000057230",
  "diseaseFromSourceId": "OMIM:606785",
  "biologicalModelAllelicComposition": "Aak1<tm1b(EUCOMM)Hmgu> hom early",
  "biologicalModelGeneticBackground": "C57BL/6N",
  "diseaseModelAssociatedModelPhenotypes": [
    {
      "id": "MP:0005567",
      "label": "decreased circulating total protein level"
    },
    {
      "id": "MP:0003795",
      "label": "abnormal bone structure"
    },
    {
      "id": "MP:0001399",
      "label": "hyperactivity"
    },
    {
      "id": "MP:0011888",
      "label": "abnormal circulating total protein level"
    },
    {
      "id": "MP:0002590",
      "label": "increased mean corpuscular volume"
    },
    {
      "id": "MP:0001569",
      "label": "abnormal circulating bilirubin level"
    },
    {
      "id": "MP:0000202",
      "label": "abnormal circulating alkaline phosphatase level"
    },
    {
      "id": "MP:0002574",
      "label": "increased vertical activity"
    },
    {
      "id": "MP:0005293",
      "label": "impaired glucose tolerance"
    },
    {
      "id": "MP:0004738",
      "label": "abnormal auditory brainstem response"
    },
    {
      "id": "MP:0005419",
      "label": "decreased circulating serum albumin level"
    },
    {
      "id": "MP:0003313",
      "label": "abnormal locomotor activation"
    },
    {
      "id": "MP:0003921",
      "label": "abnormal heart left ventricle morphology"
    },
    {
      "id": "MP:0001486",
      "label": "abnormal startle reflex"
    },
    {
      "id": "MP:0001489",
      "label": "decreased startle reflex"
    },
    {
      "id": "MP:0011960",
      "label": "abnormal eye anterior chamber depth"
    },
    {
      "id": "MP:0000558",
      "label": "abnormal tibia morphology"
    },
    {
      "id": "MP:0002764",
      "label": "short tibia"
    },
    {
      "id": "MP:0011110",
      "label": "preweaning lethality, incomplete penetrance"
    },
    {
      "id": "MP:0009141",
      "label": "increased prepulse inhibition"
    }
  ],
  "diseaseModelAssociatedHumanPhenotypes": [
    {
      "id": "HP:0008282",
      "label": "Unconjugated hyperbilirubinemia"
    }
  ],
  "biologicalModelId": "MGI:5766745"
}
```

### CODE REVIEW

File: modules/MouseModels.py [hunk 1]
```diff
@@ -728,6 +716,36 @@ def generate_phenodigm_evidence_strings(self, upper_limit=0):
...
+        This function maps the old evidence structure into the new json schema.
+
+        Args:
+            evidenceString (dict): old PhenoDigm evidence string
+        Returns:
+            dict: dictionary for the new evidence if all fields could be successfully mapped. 
+        '''
+
+        try:
+            return collections.OrderedDict({
```
Comment (ireneisdoomed) [inline modules/MouseModels.py:730]: Is it possible that in this function `data` has to be replaced by `evidenceString`?

File: modules/MouseModels.py [hunk 2]
```diff
@@ -740,8 +758,10 @@ def write_evidence_strings(self, filename):
                 error = evidenceString.validate(self._logger)
                 score = evidenceString.evidence.disease_model_association.resource_score.value
                 if error == 0 and score >= 0.9:
-                    tp_file.write(evidenceString.to_JSON(indentation=None) + "\n")
-                    countExported+=1
+                    new_evidnce = self.convert_evidence(evidenceString)
+                    if new_evidnce:
+                        tp_file.write(new_evidnce.to_JSON(indentation=None) + "\n")
```
Comment (ireneisdoomed) [inline modules/MouseModels.py:763]: Should we start outputting everything as parquet files?

File: modules/MouseModels.py [hunk 3]
```diff
@@ -832,20 +851,39 @@ def process_all(self, update_cache=False, write2cloud=False):
...
+        action="store_true", dest="write2cloud", default=False)
+
+    # log file is optional:
+    parser.add_argument("-l", '--log-file', help='Optional filename for logfile.', required=False)
+    args = parser.parse_args()
+
+    # Initialize logger:
+    # Initialize logger based on the provided logfile. 
+    # If no logfile is specified, logs are written to stderr 
+    logging.basicConfig(
```
Comment (ireneisdoomed) [inline modules/MouseModels.py:873]: Could this logger conflict with the one implemented within the class?

File: modules/MouseModels.py (hunk 1 above)
Comment (DSuveges) [reply modules/MouseModels.py:728]: Absolutely. Fixed.

File: modules/MouseModels.py (hunk 3 above)
Comment (DSuveges) [reply modules/MouseModels.py:848]: I did a bit of a cleanup in the logging. We'll see how it works.

**Review by ireneisdoomed — APPROVED** [review]
Everything looks fine. I've ran it locally without any problems.

**Pull request merged**